### PR TITLE
Fix the comment for the Job managedBy field

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind documentation
/kind api-change

#### What this PR does / why we need it:

To make the comment consistent with the actual implmenation:
- the length is 63 [link](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/validation/validation.go#L67-L68)
- the field is immutable [link](https://github.com/kubernetes/kubernetes/blob/9d87fa215d9e8020abdc17132d1252536cd752d2/pkg/apis/batch/validation/validation.go#L613)

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fix the comment for the Job's managedBy field
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/batch/v1/types.go [hunk 1]
```diff
@@ -477,7 +477,8 @@ type JobSpec struct {
 	// The value must be a valid domain-prefixed path (e.g. acme.io/foo) -
 	// all characters before the first "/" must be a valid subdomain as defined
 	// by RFC 1123. All characters trailing the first "/" must be valid HTTP Path
-	// characters as defined by RFC 3986. The value cannot exceed 64 characters.
+	// characters as defined by RFC 3986. The value cannot exceed 63 characters.
+	// The field is immutable.
```
Comment (atiratree) [inline staging/src/k8s.io/api/batch/v1/types.go:481]: nit:

```suggestion
	// This field is immutable.
```

might be better

Comment (mimowo) [reply staging/src/k8s.io/api/batch/v1/types.go:481]: Probably slightly, let me update.

Comment (mimowo) [reply staging/src/k8s.io/api/batch/v1/types.go]: Done, while doing so I realized I forgot to update ` pkg/apis/batch/types.go`. Now done.

**Comment by sttts** [comment]
This is an API change formally and needs somebody from @kubernetes/api-approvers.

**Comment by sttts** [comment]
/unassign

@liggitt @thockin

**Pull request merged**